projects/igniteui-angular/src/lib/grids/grid/column-hiding.spec.ts

```diff
@@ -588,6 +588,51 @@ describe('Column Hiding UI', () => {
             expect(getButtonDisabledState('Hide All')).toBe(false, 'Hide All is not enabled!');
         }));
 
+        it('- When Hide All columns no rows should be rendered', (async () => {
+            grid.rowSelectable = true;
+            grid.paging = true;
+            fix.detectChanges();
+
+            grid.rowDraggable = true;
+            fix.detectChanges();
+
+            grid.groupBy({
+                fieldName: 'Released', dir: SortingDirection.Desc, ignoreCase: false
+            });
+            fix.detectChanges();
+            let tHeadItems = fix.nativeElement.querySelector('igx-grid-header-group');
+            let gridRows = fix.nativeElement.querySelector('igx-grid-row');
+            let paging = fix.nativeElement.querySelector('.igx-grid-paginator');
+            let rowSelectors = grid.nativeElement.querySelector('.igx-checkbox');
+            let dragIndicators = grid.nativeElement.querySelector('.igx-grid__drag-indicator');
+            let verticalScrollBar = grid.nativeElement.querySelector('.igx-grid__tbody-scrollbar[hidden]');
+
+            expect(tHeadItems).not.toBeNull();
+            expect(gridRows).not.toBeNull();
+            expect(paging).not.toBeNull();
+            expect(rowSelectors).not.toBeNull();
+            expect(dragIndicators).not.toBeNull();
+            expect(verticalScrollBar).toBeNull();
+
+            grid.columnList.forEach((col) => col.hidden = true);
+            await wait();
+            fix.detectChanges();
+
+            tHeadItems = fix.nativeElement.querySelector('igx-grid-header-group');
+            gridRows = fix.nativeElement.querySelector('igx-grid-row');
```

Comment:
Maybe we can use GridFunctions class helper methods for getting various elements from the grid or add such methods, so we can reuse them later on.